libafl_libfuzzer/runtime/src/fuzz.rs

```diff
@@ -194,35 +215,47 @@ pub fn fuzz(
     options: &LibfuzzerOptions,
     harness: &extern "C" fn(*const u8, usize) -> c_int,
 ) -> Result<(), Error> {
+    #[cfg(unix)]
     if let Some(forks) = options.forks() {
         let shmem_provider = StdShMemProvider::new().expect("Failed to init shared memory");
+
+        #[cfg(feature = "tui_monitor")]
         if options.tui() {
             let monitor = TuiMonitor::builder()
                 .title(options.fuzzer_name())
                 .enhanced_graphics(true)
                 .build();
-            fuzz_many_forking(options, harness, shmem_provider, forks, monitor)
-        } else if forks == 1 {
-            let monitor = MultiMonitor::new(create_monitor_closure());
-            fuzz_single_forking(options, harness, shmem_provider, monitor)
-        } else {
-            let monitor = MultiMonitor::new(create_monitor_closure());
-            fuzz_many_forking(options, harness, shmem_provider, forks, monitor)
+            return fuzz_many_forking(options, harness, shmem_provider, forks, monitor);
         }
-    } else if options.tui() {
+
+        // Non-TUI path or when tui_monitor feature is disabled
+        let monitor = MultiMonitor::new(create_monitor_closure());
+
+        if forks == 1 {
+            return fuzz_single_forking(options, harness, shmem_provider, monitor);
+        }
+
+        return fuzz_many_forking(options, harness, shmem_provider, forks, monitor);
+    }
+
+    #[cfg(feature = "tui_monitor")]
+    if options.tui() {
         // if the user specifies TUI, we assume they want to fork; it would not be possible to use
         // TUI safely otherwise
         let shmem_provider = StdShMemProvider::new().expect("Failed to init shared memory");
         let monitor = TuiMonitor::builder()
             .title(options.fuzzer_name())
             .enhanced_graphics(true)
             .build();
-        fuzz_many_forking(options, harness, shmem_provider, 1, monitor)
-    } else {
-        destroy_output_fds(options);
-        fuzz_with!(options, harness, do_fuzz, |fuzz_single| {
-            let mgr = SimpleEventManager::new(MultiMonitor::new(create_monitor_closure()));
-            crate::start_fuzzing_single(fuzz_single, None, mgr)
-        })
+        return fuzz_many_forking(options, harness, shmem_provider, 1, monitor);
     }
+
+    // Default path when no forks or TUI are specified, or when tui_monitor feature is disabled
+    #[cfg(unix)]
+    destroy_output_fds(options);
+
+    fuzz_with!(options, harness, do_fuzz, |fuzz_single| {
+        let mgr = SimpleEventManager::new(MultiMonitor::new(create_monitor_closure()));
```

Comment:
LibAFL_Libfuzzer doesn't support Llmp?

Reply:
I guess it doesn't right now but no reason we couldn't add it. Could use llmp+multiprocessing on windows where we can't fork